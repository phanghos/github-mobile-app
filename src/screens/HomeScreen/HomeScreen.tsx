import React from 'react';
import { Text, TouchableOpacity, View } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import {
  ISSUES_SCREEN,
  PULL_REQUESTS_SCREEN,
  REPOSITORIES_SCREEN,
} from 'consts/navigationConsts';

const Item = ({ text, onPress }: { text: string; onPress?: () => void }) => {
  return (
    <TouchableOpacity onPress={onPress} style={{ marginVertical: 6 }}>
      <Text style={{ fontWeight: '600' }}>{text}</Text>
    </TouchableOpacity>
  );
};

export const HomeScreen = () => {
  const { navigate } = useNavigation();

  return (
    <View style={{ flex: 1, alignItems: 'center', paddingVertical: 48 }}>
      <Item text="Issues" onPress={() => navigate(ISSUES_SCREEN)} />
      <Item text="Repositories" onPress={() => navigate(REPOSITORIES_SCREEN)} />
      <Item
        text="Pull Requests"
        onPress={() => navigate(PULL_REQUESTS_SCREEN)}
      />
    </View>
  );
};
