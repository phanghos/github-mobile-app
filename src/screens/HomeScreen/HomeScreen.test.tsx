import React from 'react';
import { render } from '@testing-library/react-native';
import { HomeScreen } from './HomeScreen';

describe('HomeScreen', () => {
  test('Renders correctly', () => {
    const tree = render(<HomeScreen />);

    expect(tree).toMatchSnapshot();
  });
});
