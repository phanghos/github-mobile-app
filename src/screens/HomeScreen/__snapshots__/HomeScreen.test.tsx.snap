// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingTop": 48,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "height": 48,
          "justifyContent": "center",
          "position": "absolute",
          "width": "100%",
          "zIndex": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 48,
          "paddingLeft": 16,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          accessibilityRole="header"
          aria-level="1"
          numberOfLines={1}
          style={
            Object {
              "color": "rgb(28, 28, 30)",
              "fontFamily": "OpenSans-Bold",
              "fontSize": 17,
              "fontWeight": undefined,
            }
          }
        >
          Home
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <SvgMock
          style={
            Object {
              "marginRight": 16,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 16,
        "paddingTop": 24,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 18,
            "marginBottom": 24,
          },
          Object {
            "fontFamily": "OpenSans-SemiBold",
          },
        ]
      }
    >
      My Work
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginVertical": 8,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            undefined,
            Object {
              "fontFamily": "OpenSans-Regular",
            },
          ]
        }
      >
        Issues
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginVertical": 8,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            undefined,
            Object {
              "fontFamily": "OpenSans-Regular",
            },
          ]
        }
      >
        Pull Requests
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginVertical": 8,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            undefined,
            Object {
              "fontFamily": "OpenSans-Regular",
            },
          ]
        }
      >
        Repositories
      </Text>
    </View>
  </View>
</View>
`;
